egs/wsj0-mix/DeepClustering/train.py

```diff
@@ -0,0 +1,123 @@
+""" Train a deep clustering network
+"""
+import os
+import argparse
+from ipdb import set_trace
+
+import torch
+from torch.utils.data import DataLoader
+import pytorch_lightning as pl
+from pytorch_lightning.callbacks import ModelCheckpoint
+
+from asteroid.data.wsj0_mix import WSJ2mixDataset 
+from asteroid.engine.system import System
+from asteroid.losses.cluster import deep_clustering_loss
+import asteroid.filterbanks as fb
+from asteroid.filterbanks.inputs_and_masks import take_mag
+from model import make_model_and_optimizer
+
+EPS = torch.finfo(torch.float).eps
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--gpus', type=str, help='list of GPUs', default='-1')
+parser.add_argument('--exp_dir', default='exp/tmp',
+                    help='Full path to save best validation model')
+
+class DcSystem(System):
+    """ System for deep clustering implementation. 
+        Overwrites the common_step.
+    """
+    def __init__(self, model, optimizer, loss_func, train_loader,
+                 val_loader=None, scheduler=None, config=None):
+        System.__init__(self, model, optimizer, loss_func, train_loader,
```

Comment:
Can we use `super` here please? With python3 style `super`, no need to name the `System` class so it's easier to refactor if needed.